src/librustc/ty/mod.rs

```diff
@@ -1034,7 +1034,7 @@ impl<'tcx> ToPolyTraitRef<'tcx> for PolyProjectionPredicate<'tcx> {
         // This is because here `self` has a `Binder` and so does our
         // return value, so we are preserving the number of binding
         // levels.
-        ty::Binder(self.0.projection_ty.trait_ref)
+        ty::Binder(ty::tls::with(|tcx| self.0.projection_ty.trait_ref(tcx)))
```

Comment:
Yeah, that could do the work itself. Not sure if `data.trait_ref(tcx).to_poly_trait_ref()` works though (it might assert).